controllers/util/k8sclient.go

```diff
@@ -0,0 +1,24 @@
+package util
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func NewK8sClient(scheme *runtime.Scheme) (client.Client, error) {
```

Comment:
1. There is no need to have a new util file. This is definitely a k8s util function.
2. Moving this into util is a good move, but you still need to consolidate all the calla (for client creation) across the repo